### PR TITLE
*: remove is_cockroach special casing

### PR DESCRIPTION
**Pull request by mjibson**
Now that we use cockroach in all testing we can remove the ability to work with postgres and assume cockroach.

### Motivation

  * This PR adds a feature that has not yet been specified.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

**Comment by mjibson** [comment]
Ah, sadly many tests still use Postgres. :(

**Comment by benesch** [comment]
> Ah, sadly many tests still use Postgres. :(

Just one, really; all the "checks" tests use the same infrastructure. I'll put up a PR shortly to port that over to Cockroach.

**Comment by benesch** [comment]
As promised, here's the PR that ports the platform checks to CockroachDB: https://github.com/MaterializeInc/materialize/pull/17023.

**Comment by benesch** [comment]
Pushed up a fix to have `test_stash_postgres` use the `COCKROACH_URL` environment variable instead of the `POSTGRES_URL` environment variable.

**Comment by mjibson** [comment]
Great! Can you hit approve?

**Comment by benesch** [comment]
I was going to defer to @pH14 (or @danhhz) on approval, since I wasn't sure `persist` was bought in on losing PostgreSQL as a backend!

**Review by danhhz — COMMENTED** [review]
I'm on board with dropping postgres support, src/persist changes LGTM. Did we ever figure out the root of Frank's local dev perf issues on CRDB? Is there a real regression between the two?

**Review by benesch — APPROVED** [review]

**Comment by benesch** [comment]
> Did we ever figure out the root of Frank's local dev perf issues on CRDB? Is there a real regression between the two?

CockroachDB on Mac natively is slow; running it in Docker eliminates the perf regression. Something to look into, for sure, but IMO run-Cockroach-in-Docker is a sufficient workaround that we don't need to hold off on this.

**Comment by danhhz** [comment]
woah, TIL! that's true even for running an arm64 binary natively?

> IMO run-Cockroach-in-Docker is a sufficient workaround that we don't need to hold off on this.

👍

**Comment by mjibson** [comment]
Hmm. See https://github.com/MaterializeInc/materialize/issues/16963#issuecomment-1378273563 for more curiosities.

**Pull request merged**
